File: Algorithms-and-data-structures-master/0.10 Hash table/main.py
import numpy as np

with open("input.txt", "r") as f:
    m, c, keys_amount = map(int, f.readline().split())

    hash_table = np.full(m, -1, dtype=int)

    for _ in range(keys_amount):
        num = int(f.readline().strip())
        position = num % m
        while hash_table[position] != -1:
            position = (position + c) % m
        hash_table[position] = num

with open("output.txt", "w") as f:
    for i in range(m):
        f.write(f"{hash_table[i]} ")

File: Algorithms-and-data-structures-master/0.10 Hash table/main.cpp
#include <iostream>
#include <fstream>
#include <vector>

int main() {
    std::ifstream fin("input.txt");
    std::ofstream fout("output.txt");

    int m, c, keys_amount;
    fin >> m >> c >> keys_amount;

    std::vector<int> hash_table(m, -1);

    for (int i = 0; i < keys_amount; i++) {
        int num;
        fin >> num;
        int position = num % m;
        while (hash_table[position] != -1) {
            position = (position + c) % m;
        }
        hash_table[position] = num;
    }

    for (int i = 0; i < m; i++) {
        fout << hash_table[i] << " ";
    }

    fin.close();
    fout.close();
    return 0;
}
